src/SourceGenerators/Uno.UI.SourceGenerators/BindableTypeProviders/BindableTypeProvidersGenerationTask.cs

```diff
@@ -593,41 +593,42 @@ private bool IsOverride(IMethodSymbol? methodDefinition)
 
 			private void GenerateTypeTable(IndentedStringBuilder writer)
 			{
-				foreach (var type in _typeMap.Where(k => !k.Key.IsGenericType))
-				{
-					writer.AppendLineIndented($"private global::Uno.UI.DataBinding.IBindableType _bindableType{type.Value.Index:000};");
-				}
+				var types = _typeMap.Where(k => !k.Key.IsGenericType);
+				writer.AppendLineIndented($"private readonly global::Uno.UI.DataBinding.IBindableType[] _bindableTypes = new global::Uno.UI.DataBinding.IBindableType[{types.Count()}];");
+				writer.AppendLineIndented($"private static global::Uno.UI.DataBinding.IBindableType _null;");
 
 				using (writer.BlockInvariant("public global::Uno.UI.DataBinding.IBindableType GetBindableTypeByFullName(string fullName)"))
 				{
+					writer.AppendLineIndented("ref global::Uno.UI.DataBinding.IBindableType element = ref _null;");
```

Comment:
I'm glad to have added [validations for C# 7.3](https://github.com/unoplatform/uno/blob/bbd92e7bc7fe70e838a4eac0148e2c1e06b8b6a2/build/test-scripts/run-template-tests.ps1#L104) so that we know it still works there!